### PR TITLE
make search queue deduplication use less memory

### PR DESCRIPTION
**Pull request by nnutter**

### CODE REVIEW

File: lib/perl/Genome/Search/Queue.pm [hunk 1]
```diff
@@ -79,4 +82,70 @@ sub default_priority {
...
+              'specifies subject_id' => sub { shift->rule->value_for('subject_id') },
+              'specifies subject_class' => sub { shift->rule->value_for('subject_class') },
+          },
+        },
+    );
+
+    my $m_iter = $set->member_iterator;
+    my $q = $m_iter->next;
+    my $duplicate_count = 0;
+    while (my $q = $m_iter->next) {
```
Comment (tmooney) [inline lib/perl/Genome/Search/Queue.pm]: It strikes me as odd that both the one to keep and the ones to delete are kept in `my $q`.

Comment (nnutter) [inline lib/perl/Genome/Search/Queue.pm]: I've pushed a commit to address this.

**Comment by tmooney** [comment]
+1 :+1:

**Comment by ebelter** [comment]
+1

**Pull request merged**